_posts/2018-06-11-using-strimzi-with-local-storage.md

```diff
@@ -0,0 +1,134 @@
+It has to resync only the records which it missed during the failover - this is exactly the same as when you use persistent network attached storage.
+
+![Local persistent storage during node restart]({{ "/assets/2018-06-11-using-strimzi-with-local-storage-local-storage.png" | absolute_url }})
+
+Now ... what if your node is gone and will never return?
+In that case, you will need to make sure that the broker is scheduled on another node.
+Since the persistent volume was only local, it will be lost together with the node.
+Therefore the broker has to start from scratch and resync all data from another replica.
+This is the same as the local ephemeral storage case.
+
+# Local storage support in Strimzi
+
+Strimzi is able to provision the storage for Kafka brokers in two forms:
+
+* Ephemeral storage using the `emptyDir` volumes
+* Persistent storage using Persistent Volume Claims
+
+Persistent Volume Claims use Storage Classes to dynamically create or assign persistent volumes and bind them to the Persistent Volume Claim.
+This mechanism is today used to provision networked attached storage and use them with Kafka.
+But the Local storage type is also using the Storage Class to do the binding.
+Strimzi allows you to specify which Storage Class should be used to provision the storage.
+So by specifying the right Storage Class you can use Local type volumes in Strimzi already today.
+The example below shows the `kafka-storage` configuration using the Local persistent volumes in my Kubernetes cluster:
+
+```yaml
+kafka-storage: |-
+  {
+    "type": "persistent-claim",
+    "size": "1000Gi",
+    "class": "local-storage"
+  }
+```
+
+# When should Local storage be used
+
+Local storage is definitely an important improvement to Kubernetes.
+But is it really a good idea to use it for Kafka?
+
+When you are running your Kafka cluster on one of the three major public clouds providers, it might not be interesting.
+Regardless of whether you use Amazon AWS, Microsoft Azure or Google Cloud Platform, their network attached storage is able to deliver very good performance and their virtual machines are optimized for using it.
+On the other hand, machines with fast and large local storage are not that common and are not always cheaper than using network attached storage.
+Also the nodes of your Kubernetes cluster also tend to be treated differently in public cloud.
+Instead of doing regular maintenance and restarting the nodes, it is often easier to just spin a new fresh node and kill the old one.
+And as explained in previous chapters, this approach works perfectly with the networked attached storage.
+But doesn't fit well with the Local storage, which will be lost during such operations.
+
+On the other hand, when you run Kafka on Kubernetes or OpenShift installed _on-premise_ in your own datacenter, the Local storage might be a real gamechanger.
+You will not need to invest in high performance storage network.
```

Comment:
"high performance network storage" ? or "high performance network attached storage" ?

Reply:
This is intentional ... the network is usually the problem, not the NAS it self.